### PR TITLE
refactor(composer-app): Activate service credentials

### PR DESCRIPTION
**Pull request by richburdon**

### CODE REVIEW

File: packages/apps/composer-app/src/plugins/welcome/WelcomePlugin.tsx [hunk 1]
```diff
@@ -95,43 +89,18 @@ export const WelcomePlugin = (): PluginDefinition<SurfaceProvides & Translations
       }
 
       // Existing beta users should be able to get the credential w/o a magic link.
-      if (identity) {
+      if (hubUrl && identity && token) {
```
Comment (wittjosiah) [inline packages/apps/composer-app/src/plugins/welcome/WelcomePlugin.tsx:92]: This needs to run even without a token to migrate identities over

File: packages/ui/react-ui-theme/src/config/colors.ts [hunk 2]
```diff
@@ -225,6 +225,10 @@ export const semanticColors = {
     light: physicalColors.neutral['700'],
     dark: physicalColors.neutral['300'],
   },
+  error: {
```
Comment (wittjosiah) [inline packages/ui/react-ui-theme/src/config/colors.ts:228]: @thure does this already exist somewhere?

File: packages/apps/composer-app/src/plugins/welcome/credentials/header-codec.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+export type EncodedTimestamp = { __t: number };
+
+export interface KeyCodec<KeyType> {
+  isKey(value: any): value is KeyType;
+  encode(key: KeyType): EncodedKey;
+  decode(value: EncodedKey): KeyType;
+}
+
+export class HeaderCodec<Key> {
```
Comment (dmaretskyi) [inline packages/apps/composer-app/src/plugins/welcome/credentials/header-codec.ts:21]: use `deepMapValues` from @dxos/util -- it handles recursion for you

File: packages/ui/react-ui-theme/src/config/colors.ts (hunk 2 above)
Comment (thure) [reply packages/ui/react-ui-theme/src/config/colors.ts:228]: There is a fragment `messageValence` that could be refactored to use semantic tokens, however I have a better system ready to go with `@ch-ui/tokens` which would make both of these approaches obsolete. Basically this adds tech debt but not much.

**Pull request merged**